test/device_store_test.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:async';
+
+import 'package:firmware_updater/device_store.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:fwupd/fwupd.dart';
+import 'package:mockito/mockito.dart';
+
+import 'test_utils.dart';
+
+void main() {
+  test('monitors updatable devices', () async {
+    final deviceAdded = StreamController<FwupdDevice>.broadcast(sync: true);
+    final deviceRemoved = StreamController<FwupdDevice>.broadcast(sync: true);
+
+    final service = mockService();
+    when(service.deviceAdded).thenAnswer((_) => deviceAdded.stream);
+    when(service.deviceRemoved).thenAnswer((_) => deviceRemoved.stream);
+
+    final store = DeviceStore(service);
+    await store.init();
+    expect(store.devices, isEmpty);
+    verify(service.getDevices()).called(1);
+
+    final dev1 = FwupdDevice(
+      deviceId: '1',
+      name: '',
+      plugin: '',
+      flags: const {FwupdDeviceFlag.updatable},
+    );
+    final dev2 = FwupdDevice(
+      deviceId: '2',
+      name: '',
+      plugin: '',
+      flags: const {FwupdDeviceFlag.updatableHidden},
+    );
+    final dev3 = FwupdDevice(
+      deviceId: '2',
```

Comment:
Completely irrelevant, but `dev3` should probably have `deviceId: '3'`

Reply:
would've been fun if the test started failing after fixing the id